overlord/devicestate/devicestate.go

```diff
@@ -706,9 +707,14 @@ func remodelTasks(ctx context.Context, st *state.State, current, new *asserts.Mo
 	}
 	snapsAccountedFor[new.Gadget()] = true
 
-	// go through all the model snaps, see if there are new required snaps
-	// or a track for existing ones needs to be updated
-	for _, modelSnap := range new.SnapsWithoutEssential() {
+	// go through the model snaps not accounted for yet, see if there are
+	// new required snaps or a track for existing ones needs to be updated
+	for _, modelSnap := range new.Snaps() {
+		switch modelSnap.SnapType {
+		case "gadget", "kernel", "base", "core":
```

Comment:
Would
```go
if snapsAccountedFor[modelSnap.Name] {
	continue
}
```
work instead of specifically calling out out the already handled ones? It looks like `new.Base()`, which is used as a key in `snapsAccountedFor` earlier, returns something that may not be the name, so it might not. If it does, though, I think it'd be a bit better.

Reply:
Actually this is an excellent suggestion, as now I notice that this could overlook non-rootfs bases (we can have UC22 but also a core20 snap installed).